### PR TITLE
[Core] Small clean up of `ParallelFillCommunicator`. Also adding test

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Small clean up of  `ParallelFillCommunicator` including some missing documentation. Also adding test.

**🆕 Changelog**

- [Small clean up of `ParallelFillCommunicator`](https://github.com/KratosMultiphysics/Kratos/commit/87e8cc72d4750fdcfb447b430a2904394073a83e)
- [Adding `ParallelFillCommunicatorExecute` test](https://github.com/KratosMultiphysics/Kratos/pull/11311/commits/5c8601bbf1fac75fbba6bdf513d0af3b39f0c654)

### CODE REVIEW

File: kratos/mpi/utilities/parallel_fill_communicator.h [hunk 1]
```diff
@@ -43,14 +42,19 @@ namespace Kratos
 ///@name Kratos Classes
 ///@{
 
-class ModelPart;
+class ModelPart; // forward declaration
```
Comment (philbucher) [inline kratos/mpi/utilities/parallel_fill_communicator.h]: since we decided to not generally use forward declarations, can you please include it here?

**Review by philbucher — COMMENTED** [review]
minor, then I will approve

File: kratos/mpi/utilities/parallel_fill_communicator.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/mpi/utilities/parallel_fill_communicator.h]: This was already here, I think it is a bad idea to abandon the forward declaration

Comment (loumalouomega) [reply kratos/mpi/utilities/parallel_fill_communicator.h]: It is already included in base class, removing forward declaration

**Review by philbucher — APPROVED** [review]

**Pull request merged**